### PR TITLE
cephadm: log files created by logrotate should be owned by ceph user/group vs root

### PR DESCRIPTION
**Pull request by cfsnyder**
The logrotate configuration that is created by cephadm creates new log files with root/root ownership.
Log files should be created with Ceph user/group ownership (167/167).

Fixes: https://tracker.ceph.com/issues/54020
Signed-off-by: Cory Snyder <csnyder@iland.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/cephadm/cephadm [hunk 1]
```diff
@@ -3348,13 +3348,13 @@ def install_base_units(ctx, fsid):
     compress
     sharedscripts
     postrotate
-        killall -q -1 ceph-mon ceph-mgr ceph-mds ceph-osd ceph-fuse radosgw rbd-mirror cephfs-mirror || pkill -1 -x 'ceph-mon|ceph-mgr|ceph-mds|ceph-osd|ceph-fuse|radosgw|rbd-mirror|cephfs-mirror' || true
+        chown 167:167 /var/log/ceph/%s/*.log; killall -q -1 ceph-mon ceph-mgr ceph-mds ceph-osd ceph-fuse radosgw rbd-mirror cephfs-mirror || pkill -1 -x 'ceph-mon|ceph-mgr|ceph-mds|ceph-osd|ceph-fuse|radosgw|rbd-mirror|cephfs-mirror' || true
```
Comment (sebastian-philipp) [inline src/cephadm/cephadm:3351]: you think we can avoid hardcoding 167 here`

Comment (cfsnyder) [reply src/cephadm/cephadm:3351]: are you thinking we just add them as constants at the top of the file?

Comment (sebastian-philipp) [reply src/cephadm/cephadm:3351]: is used in some places:

https://github.com/ceph/ceph/blob/3e23850bafa4ac48bb2f2f657fca19541540a511/src/cephadm/cephadm#L2788

Comment (cfsnyder) [reply src/cephadm/cephadm]: nice, thanks @sebastian-philipp! I updated it to use this

**Review by adk3798 — APPROVED** [review]

**Comment by cfsnyder** [comment]
Actually I think we can close this one @adk3798. With further investigation, we found that logrotate uses the same uid/guid as the previous log file that it is rotating. The actual issue was that rgw op log files were being created with root ownership and couldn't be reopened after rgw dropped root privileges. This shouldn't be necessary if all log files are created properly in the first place.

**Comment by adk3798** [comment]
> Actually I think we can close this one @adk3798. With further investigation, we found that logrotate uses the same uid/guid as the previous log file that it is rotating. The actual issue was that rgw op log files were being created with root ownership and couldn't be reopened after rgw dropped root privileges. This shouldn't be necessary if all log files are created properly in the first place.

alright, I'll close this. Feel free to re-open if it tuns out to be needed

**Pull request closed without merging**